docs/sharding.md

```diff
@@ -0,0 +1,42 @@
+# Multiple Azure SignalR Service Instances Support
```

Comment:
briefly describe what is multi endpoint scenario and what is sharding for, add doc link for this concept